src/core/iomgr/tcp_server.h

```diff
@@ -80,4 +80,8 @@ int grpc_tcp_listener_get_port(grpc_tcp_listener *listener);
 void grpc_tcp_listener_ref(grpc_tcp_listener *listener);
 void grpc_tcp_listener_unref(grpc_tcp_listener *listener);
 
+/* potentially (trivially) implementable: */
+int grpc_tcp_listener_get_fd(grpc_tcp_listener *listener, unsigned index);
```

Comment:
Then the sibling fd's become un-accessible. Given that we create siblings
because a single add_port caused multiple listening sockets to be created
(due to ipv6 dual stack for example), it seems like a general API would
need this.

On Thu, Jan 7, 2016 at 5:06 PM Dan Born <notifications@github.com> wrote:

> In src/core/iomgr/tcp_server.h
> <https://github.com/grpc/grpc/pull/4555#discussion_r49148448>:
>
> > @@ -80,4 +80,8 @@ int grpc_tcp_listener_get_port(grpc_tcp_listener *listener);
> >  void grpc_tcp_listener_ref(grpc_tcp_listener *listener);
> >  void grpc_tcp_listener_unref(grpc_tcp_listener *listener);
> >
> > +/* potentially (trivially) implementable: */
> > +int grpc_tcp_listener_get_fd(grpc_tcp_listener *listener, unsigned index);
>
> Why is there an index parameter? Even if the listener has siblings, I
> think only the listener->fd should be returned here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/4555/files#r49148448>.
>